### PR TITLE
Deduct ingredients branch

### LINKED ISSUE #80: As a user, I can deduct the ingredients that I have from the recipe that I want to cook so that I save time from deleting manually.
CookRecipe Class: 
- Deducts the ingredients according to their recency of expiry

### LINKED ISSUE #108: As a user, I can deduct the ingredients that expire first so that I do not waste my ingredients.
CookRecipe Class:

Deducts the ingredients according to their recency of expiry

### PR DESCRIPTION
**Pull request by yantingsanity**
The following items has been done:
- Delete ingredients that are listed in the recipe after users typed in `cookrecipe` from the ingredient inventory 

This closes #80 and closes #108.

### CODE REVIEW

File: src/main/java/seedu/kitchenhelper/parser/Parser.java [hunk 1]
```diff
@@ -213,6 +213,7 @@ private Command prepareCookRecipe(String attributes) {
             cookCmd.setRecipeName(recipeName);
             cookCmd.setRecipePax(numOfPax);
         } catch (IndexOutOfBoundsException e) {
+            System.out.println("hi");
```
Comment (jinfayap) [inline src/main/java/seedu/kitchenhelper/parser/Parser.java]: Perhaps removing this statement for debugging.

File: src/main/java/seedu/kitchenhelper/command/CookRecipeCommand.java [hunk 2]
```diff
@@ -34,16 +35,16 @@ public String cookRecipe(ArrayList<Ingredient> ingredientList, ArrayList<Recipe>
             throws KitchenHelperException {
         // checks if the specified recipe given by user exists
         int indexOfRecipe = checkIfRecipeExist(recipeList);
-        if (indexOfRecipe != recipeList.size()) {
-            throw new KitchenHelperException("The given recipe name does not exist");
+        if (indexOfRecipe > recipeList.size()) {
+            return "The given recipe name does not exist!";
```
Comment (jinfayap) [inline src/main/java/seedu/kitchenhelper/command/CookRecipeCommand.java]: Perhaps you can use non-magic string literals?

File: src/main/java/seedu/kitchenhelper/command/CookRecipeCommand.java [hunk 3]
```diff
@@ -34,16 +35,16 @@ public String cookRecipe(ArrayList<Ingredient> ingredientList, ArrayList<Recipe>
...
         }
         System.out.println("Kitchen Helper is trying to cook!");
         Recipe recipeToBeCooked = recipeList.get(indexOfRecipe - 1);
 
         if (checkForSufficientIngredient(ingredientList, recipeToBeCooked)) {
             deductIngredients(ingredientList, recipeToBeCooked);
         } else {
-            throw new KitchenHelperException("There are insufficient/ missing ingredients!");
+            return "There are insufficient/ missing ingredients!";
```
Comment (jinfayap) [inline src/main/java/seedu/kitchenhelper/command/CookRecipeCommand.java]: Perhaps you can use non-magic string literals?

**Review by jinfayap — COMMENTED** [review]
LGTM after the minor fixes

File: src/main/java/seedu/kitchenhelper/command/CookRecipeCommand.java (hunk 3 above)
Comment (yantingsanity) [reply src/main/java/seedu/kitchenhelper/command/CookRecipeCommand.java]: Thank you for the feedback, I have edited some of them into non-magic string literals!

**Pull request merged**